[COPS-7154] 2.1.1 Air-Gapped Bundles v2 (#4220)
mesosphere/dcos-docs-site: pages/dkp/konvoy/2.1/choose-infrastructure/airgapbundle/loadbootstrap/index.md

---
layout: layout.pug
navigationTitle: Load Bootstrap Image
title: Load Bootstrap Image
menuWeight: 10

> excerpt: Setup docker in your environment, configure your local registry, and load the bootstrap image.

Reviewer: ```suggestion
excerpt: Set up docker in your environment, configure your local registry, and load the bootstrap image.
```

beta: false
enterprise: false
---
In order to use the DKP CLI, you'll have to do the following in sequential order:

* Install the Docker engine
* Load and create a local registry
* Update Docker configuration with registry reference
* Seed a local registry
* Load the bootstrap image

## Prerequisites

* Docker CE requires a minimum version of 19.03

> ## Setup docker

Reviewer: ```suggestion
## Set up docker
```

```bash
cat <<-EOF | sudo tee /etc/docker/daemon.json
{
  "insecure-registries" : ["$(hostname -I | awk '{print $1}'):5000"]
}
EOF
```

1. Restart docker.

```bash
systemctl restart docker
```

1. Load and start the registry image.

```bash
docker load < registry_2.tar
docker run -d --name=registry --restart=always -p 5000:5000 registry:2
```
1. Seed the local or private registry by executing the following script using the provided image bundles.

```bash
cat << seed.sh > EOF
AIRGAPPED_TAR_FILE="$@"
echo "Loading ${AIRGAPPED_TAR_FILE}"
docker load <"${AIRGAPPED_TAR_FILE}"

while read -r IMAGE; do
    echo "Processing ${IMAGE}"
    REGISTRY_IMAGE="$(echo "${IMAGE}" | sed -E "s@^(quay|k8s\.gcr|mcr\.microsoft|us\.gcr|nvcr|gcr|ghcr|docker).(io|com)@${REGISTRY_URL}@")"
    echo "Tag ${REGISTRY_IMAGE}"
    docker tag "${IMAGE}" "${REGISTRY_IMAGE}"
    docker push "${REGISTRY_IMAGE}"
done < <(tar xfO "${AIRGAPPED_TAR_FILE}" "index.json" | grep -oP '(?<="io.containerd.image.name":").*?(?=",)')

echo "Loaded ${AIRGAPPED_TAR_FILE}"
EOF

seed.sh ${KOMMANDER_AIRGAPPED_TAR_FILE}
```
## Load the bootstrap image and verify the contents

> 1. Once the registry has been seeded, load the bootstrap image.

Reviewer: ```suggestion
1. Load the bootstrap image, after the registry is seeded.
```
